### PR TITLE
Fix failing cart e2e test: fix setup wizard logic if new onboarding is presented

### PR DESCRIPTION
**Pull request by juliaamosova**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Cart e2e test (`tests/e2e-tests/specs/front-end/front-end-cart.test.js`) fails randomly right now with the following message:

```
FAIL  tests/e2e-tests/specs/front-end/front-end-cart.test.js (31.921s)
  Cart page
    ✕ should display no item in the cart (2350ms)
    ✕ should add the product to the cart when "Add to cart" is clicked (2867ms)
    ✕ should increase item qty when "Add to cart" of the same product is clicked (2332ms)
    ✕ should update qty when updated via qty input (1069ms)
    ✕ should remove the item from the cart when remove is clicked (1319ms)
    ✕ should update subtotal in cart totals when adding product to the cart (2320ms)
    ✕ should go to the checkout page when "Proceed to Checkout" is clicked (1864ms)

  ● Cart page › should display no item in the cart

    TimeoutError: Element .cart-empty (text: "Your cart is currently empty.") not found

      waiting for function failed: timeout 500ms exceeded

  ● Cart page › should add the product to the cart when "Add to cart" is clicked

    expect(received).resolves.toHaveLength(expected)

    Expected length: 1
    Received length: 0
    Received array:  []

      173 | 		const cartItemXPath = getCartItemExpression( productTitle, cartItemArgs );
      174 |
    > 175 | 		await expect( page.$x( cartItemXPath ) ).resolves.toHaveLength( 1 );
          | 		                                                  ^
      176 | 	},
      177 |
      178 | 	fillBillingDetails: async (	customerBillingDetails ) => {
```

In my troubleshooting, I discovered that the issue is actually not with the `cart` test but with the `setup wizard` test. When the new onboarding is present, we previously selected to go through the old onboarding in the test but there was no further step to actually complete the old onboarding when it was selected over the new one. As a result, the onboarding would not be completed and the cart page not created. The `cart` test that goes after the `setup wizard` test would fail because there is no cart page created. I noticed other tests were failing as well (not just the `cart` test) because of the exact same reason. 

This also explains why it was failing randomly - the new onboarding is not always present (currently only for 10% of the users randomly). 

### How to test the changes in this Pull Request:

To properly test it, you'd need to test with both new onboarding present and not. Once you built the test site but before you run the tests:

1. Activate new onboarding by using the following [snippet](https://gist.github.com/justinshreve/6e945dcb4714134156aba1b3c7f086cf); 
2. Run `npm run test:e2e` and make sure all tests pass;
3. Run `npm run test:e2e-dev` and make sure all tests pass.

Next, rebuild the test site by running the following commands:

1. Press `Ctrl+C` in the terminal window where the containers are running;
2. Stop the container(s) and delete all volumes using the following command: `docker-compose down -v`;
3. Restart the containers using the following command: `docker-compose up --build`.

Once the new site is built, test with the new onboarding off:

1. Deactivate new onboarding by using the following [snippet](https://gist.github.com/justinshreve/6e945dcb4714134156aba1b3c7f086cf) but change `b` to `a` instead (on line 10 of the snippet);
2. Run `npm run test:e2e` and make sure all tests pass;
3. Run `npm run test:e2e-dev` and make sure all tests pass.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

N/A

### CODE REVIEW

File: tests/e2e-tests/specs/activate-and-setup/setup-wizard.test.js [hunk 1]
```diff
@@ -60,148 +61,12 @@ describe( 'Store owner can go through store Setup Wizard', () => {
...
-				] );
-
-				// Skip activate Jetpack section
-				// Click on "Skip this step" in order to skip Jetpack installation
-				await page.click( '.wc-setup-footer-links' );
-
-				// Finish Setup Wizard - Ready! section
-				// Visit Dashboard
-				await StoreOwnerFlow.openDashboard();
+				await completeOldSetupWizard();
```
Comment (peterfabian) [inline tests/e2e-tests/specs/activate-and-setup/setup-wizard.test.js]: I think the `else` branch here is superfluous, it seems to me that the `completeOldSetupWizard` effectively runs in either case after the `if ( newSetupWizardNotice ) {}` block.

File: tests/e2e-tests/utils/components.js [hunk 2]
```diff
@@ -23,6 +23,149 @@ const verifyAndPublish = async () => {
 	await expect( page ).toMatchElement( '.updated.notice', { text: 'Product published.' } );
 };
 
+/**
+ * Complete old setup wizard.
+ */
+const completeOldSetupWizard = async () => {
```
Comment (peterfabian) [inline tests/e2e-tests/utils/components.js]: Nice that you factored the code out to the function 👍

**Review by peterfabian — COMMENTED** [review]
Works fine, all tests pass 👍 Nice work digging into it.

Just one small coding style suggestion and we're good to go.

File: tests/e2e-tests/specs/activate-and-setup/setup-wizard.test.js (hunk 1 above)
Comment (juliaamosova) [reply tests/e2e-tests/specs/activate-and-setup/setup-wizard.test.js]: Thanks for noticing it Peter! You are right. I removed the unnecessary `else` statement.

**Review by peterfabian — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**